### PR TITLE
Do not force to match the last measurement

### PR DESCRIPTION
**Pull request by ptpt**
See https://github.com/valhalla/valhalla/pull/865#issuecomment-319695229

### CODE REVIEW

**Comment by kevinkreiser** [comment]
@ptpt can you add the test case from my other pr, to this?

File: src/meili/map_matcher.cc [hunk 1]
```diff
@@ -293,8 +335,7 @@ MapMatcher::OfflineMatch(
   auto latest_match_measurement = begin;
   for (auto measurement = std::next(begin); measurement != end; measurement++) {
     const auto sq_distance = GreatCircleDistanceSquared(*latest_match_measurement, *measurement);
-    // Always match the last measurement
-    if (sq_interpolation_distance < sq_distance || std::next(measurement) == end) {
+    if (sq_interpolation_distance < sq_distance) {
```
Comment (ptpt) [inline src/meili/map_matcher.cc]: Note this change @kevinkreiser

**Comment by ptpt** [comment]
I got this nice visualization of the failing test http://bl.ocks.org/d/c48d41a61d5e64c299f7667339ae3df0



Don't know what the max interpolation distance is. I guess that the last measurements are interpolated on the red segment. In this case, it seems that it's still better to match the last measurement...

**Comment by kevinkreiser** [comment]
the interpolation distance for those tests is set to 10 meters so maybe, because the input points are clearly further apart than that, there is some other bug in your changes?

File: src/meili/map_matcher.cc (hunk 1 above)
Comment (kevinkreiser) [inline src/meili/map_matcher.cc]: yep that is what i was expecting you to change

**Comment by ptpt** [comment]
The last two measurements are closer than 10 meters (it is 9.44 meters). The second last measurement is preferred to match the red road segment instead of the green segment due to the turn penalty to the green segment.

**Comment by ptpt** [comment]
Close because it failed to pass the tests :(

**Pull request closed without merging**